lib/models/project.js

```diff
@@ -625,14 +620,18 @@ Project.closestSync = function(pathName, _ui, _cli) {
   @static
   @method projectOrnullProject
   @param  {UI} _ui The UI instance to provide to the created Project.
+  @param  {CLI} _cli The CLI instance to provide to the created Project.
   @return {Project}         Project instance
  */
-Project.projectOrnullProject = function(_ui, _cli) {
+Project.projectOrnullProject = function(ui, cli) {
+  this.ui = ui;
```

Comment:
`this` is actually `Project` itself here, not `Project.prototype`. We should not put `state` such as `ui` on the `Project` class itself.